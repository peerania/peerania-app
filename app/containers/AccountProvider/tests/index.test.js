import React from 'react';
import { render } from '@testing-library/react';
// import { mount } from 'enzyme';

import { AccountProvider } from '../index';

const props = {
  getCurrentAccountDispatch: jest.fn(),
};

describe('<AccountProvider />', () => {
  it('componentDidMount', async () => {
    await cmp.componentDidMount();
    expect(props.getCurrentAccountDispatch).toHaveBeenCalled();
  // it('componentDidMount', () => {
  //   AccountProvider.prototype.getCurrentAccountDispatch = jest.fn();
  //   const spy = jest.spyOn(
  //     AccountProvider.prototype,
  //     'getCurrentAccountDispatch',
  //   );
  //   const wrapper = mount(
  //     <AccountProvider {...props}>
  //       <div />
  //     </AccountProvider>,
  //   );
  //   wrapper.instance().getCurrentAccountDispatch();
  //   expect(spy).toHaveBeenCalled();
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <AccountProvider {...props}>
        <div />
      </AccountProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
