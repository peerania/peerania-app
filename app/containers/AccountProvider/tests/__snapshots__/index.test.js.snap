// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountProvider /> should render and match the snapshot 1`] = `<div />`;
